runtime/Cpp/runtime/src/UnbufferedCharStream.cpp

```diff
@@ -101,7 +98,7 @@ size_t UnbufferedCharStream::LA(ssize_t i) {
     return EOF;
   }
 
-  if (_data[static_cast<size_t>(index)] == 0xFFFF) {
+  if (_data[static_cast<size_t>(index)] == WEOF) {
```

Comment:
What is WEOF?

Reply:
WEOF is the wide character version of EOF, defined in wchar.h. I switched it to `std::char_traits<wchar_t>::eof()` which is the preferred C++ equivalent.